src/extension/src/ActionHandler.py

```diff
@@ -71,9 +90,20 @@ def update(self):
          on uninstall the agent deletes removes configuration files"""
 
         # config folder path is usually something like: /var/lib/waagent/Microsoft.CPlat.Core.LinuxPatchExtension-<version>/config
+        stdout_file_mirror = None
+        file_logger = None
         try:
+            log_file_name = datetime.datetime.utcnow().strftime("%Y-%m-%d_%H-%M-%S") + "_Update"
+            file_logger = self.utility.create_log_file(self.ext_env_handler.log_folder, log_file_name)
+            if file_logger is not None:
+                self.logger.file_logger = file_logger
+                stdout_file_mirror = StdOutFileMirror(file_logger)
+
             self.logger.log("Extension is being updated to the latest version. Copying the required extension artifacts from preceding version to the current one")
 
+            seq_no = self.ext_config_settings_handler.get_seq_no(is_enable_request=False)
+            #toDo: use seq_no to create status file to log errors
+
```

Comment:
Break this out into a second review (branch from this branch if required for work in progress). Makes it easier to catch if there are any oversights by keeping each review (+ iterations) simple.